packages/database/src/core/util/ServerValues.ts

```diff
@@ -48,13 +49,62 @@ export const generateWithValues = function(
  */
 export const resolveDeferredValue = function(
   value: { [k: string]: any } | string | number | boolean,
+  existing: Node,
   serverValues: { [k: string]: any }
 ): string | number | boolean {
   if (!value || typeof value !== 'object') {
     return value as string | number | boolean;
+  }
+  assert('.sv' in value, 'Unexpected leaf node or priority contents');
+
+  if (typeof value['.sv'] === 'string') {
+    return resolveScalarDeferredValue(value['.sv'], existing, serverValues);
+  } else if (typeof value['.sv'] === 'object') {
+    return resolveComplexDeferredValue(value['.sv'], existing, serverValues);
+  } else {
+    assert(false, 'Unexpected server value: ' + JSON.stringify(value, null, 2));
```

Comment:
Can we just throw an Error here? We don't yet use the asserts from ./util in database.

Reply:
Oh, sorry, I looked at your PR in an IDE with no diffbase. You can leave it as is.

Reply:
ok